.github/workflows/gh-test.yml

```diff
@@ -0,0 +1,65 @@
+name: Test cunumeric on GH
+
+on:
+  workflow_call:
+    inputs:
+      name:
+        required: true
+        type: string
+      build-target:
+        required: true
+        type: string
+      runs-on:
+        required: true
+        type: string
+      has-gpu:
+        required: true
+        type: boolean
+        description: "The runner has GPU(s)."
+      sha:
+        required: true
+        type: string
+      test-options:
+        required: true
+        type: string
+      log-name:
+        required: true
+        type: string
+      enabled:
+        required: true
+        type: boolean
+
+jobs:
+  test:
+    name: ${{ inputs.name }}
+    if: inputs.enabled && github.repository_owner == 'nv-legate'
+    runs-on: ${{ inputs.runs-on }}
+    container:
+      options: -u root
```

Comment:
Why do we need root?

Reply:
Because the runner user that owns the host volume mount isn't a standard UID (like 1000), so the container's non-root user can't necessarily access the files in `${{ github.workspace }}`. We have to be root in order to move files around (and/or chown the mount to the container's non-root `coder` user), and then run each step as the non-root user as necessary.